lib/topological_inventory/providers/common/operations/sources_api_client.rb

```diff
@@ -25,6 +25,11 @@ def fetch_default_endpoint(source_id)
             endpoints.find(&:default)
           end
 
+          def fetch_application(source_id)
+            applications = api.list_source_applications(source_id)&.data || []
+            applications.first
```

Comment:
as we are checking first endpoint (as default), there is no such concept in applications. 
shouldn't we check all of them?

Reply:
Hmmm yeah good question, should we fail the whole check if one of them is unavailable? Or only fail if _all_ of them are unavailable?

Reply:
yeah, also good questions. This is the "Partially available" problem. @gmcculloug ?

Reply:
For help on the original thoughts. Here is the Sources Availability matrix put together by Richard: https://github.com/RedHatInsights/sources-ui/issues/198